pkg/schema/schema.go

```diff
@@ -97,8 +102,11 @@ func NewMapItemType(item *yamlmeta.MapItem) (*MapItemType, error) {
 	if err != nil {
 		return nil, err
 	}
-
-	return &MapItemType{Key: item.Key, ValueType: typeOfValue, defaultValue: typeOfValue.GetDefaultValue(), Position: item.Position}, nil
+	defaultValue, err := getValue(item, typeOfValue)
+	if err != nil {
+		return nil, err
+	}
+	return &MapItemType{Key: item.Key, ValueType: typeOfValue, defaultValue: defaultValue, Position: item.Position}, nil
```

Comment:
nit: let's format this function just like the other `New<NodeType>Type`:

```suggestion
	
	defaultValue, err := getValue(item, typeOfValue)
	if err != nil {
		return nil, err
	}
	
	return &MapItemType{Key: item.Key, ValueType: typeOfValue, defaultValue: defaultValue, Position: item.Position}, nil
```

And the whitespace format seems easiest to read for me.